admin-dev/themes/new-theme/js/pages/carrier/add-range-handler.js

```diff
@@ -0,0 +1,143 @@
+    this.rangeRemovingBtnRow = rangeRemovingBtnRow;
+    this.removeRangeBtn = removeRangeBtn;
+    this.zoneCheckbox = zoneCheckbox;
+
+    this.$addRangeBtn = $(addRangeBtn);
+    this.$rows = $(`${rangesTable} tr:not(${rangeRemovingBtnRow})`);
+    this.$rangePriceTemplate = $(rangePriceTemplate);
+    this.$rangeFromTemplate = $(rangeFromTemplate);
+    this.$rangeToTemplate = $(rangeToTemplate);
+
+    this.handle();
+
+    return {};
+  }
+
+  /**
+   * Initiates the handler
+   */
+  handle() {
+    this.$addRangeBtn.on('click', () => {
+      this.addRangeColumn();
+    });
+
+    $(document).on('click', this.removeRangeBtn, (event) => {
+      this.removeRangeColumn(event);
+    });
+  }
+
+  /**
+   * Add new column of inputs to table
+   */
+  addRangeColumn() {
+
+    for (let i = 0; i < Object.keys(this.$rows).length; i++) {
+      const $row = $(this.$rows[i]);
+
+      if ($row.hasClass('js-range-from')) {
+
+        // replace range-from index placeholder with actual value and enable input
+        const inputFrom = (this.$rangeFromTemplate.get(0).outerHTML)
+          .replace(/__RANGE_INDEX__/g, this.rangeIndex)
+          .replace(/disabled=""/, '');
+
+        // append table data with input from template
+        $row.append(`<td data-range-index="${this.rangeIndex}">${inputFrom}</td>`);
+      } else if ($row.hasClass('js-range-to')) {
+
+        // replace range-to index placeholder with actual value and enable input
+        const inputTo = (this.$rangeToTemplate.get(0).outerHTML)
+          .replace(/__RANGE_INDEX__/g, this.rangeIndex)
+          .replace(/disabled=""/, '');
+
+        // append table data with input from template
+        $row.append(`<td data-range-index="${this.rangeIndex}">${inputTo}</td>`);
+      } else {
+        // replace price index and zone id placeholders with actual values and enable input
+        let inputPrice = (this.$rangePriceTemplate.get(0).outerHTML)
+          .replace(/__RANGE_INDEX__/g, this.rangeIndex)
+          .replace(/disabled=""/, '')
+          .replace(/__ZONE_ID__/g, $row.data('zone-id'));
+
+        // check if corresponding zone is checked to remove readonly attr
+        if ($row.find(this.zoneCheckbox).is(':checked')) {
+          inputPrice = inputPrice.replace(/readonly=""/, '');
+        }
+
+        // append table data with input from template
+        $row.append(`<td data-range-index="${this.rangeIndex}">${inputPrice}</td>`);
+      }
+    }
+
+    this.appendRangeRemovalButton(this.rangeIndex);
+
+    // increment rangeIndex value by one to keep range identification unique
+    this.rangeIndex += 1;
+  }
+
+  /**
+   * Append button which removes corresponding range column
+   *
+   * @param rangeIndex
+   */
+  appendRangeRemovalButton(rangeIndex) {
+    $(this.rangeRemovingBtnRow).removeClass('d-none');
+    $(this.rangeRemovingBtnRow).append(
+      `<td data-range-index="${rangeIndex}">${$(this.removeRangeBtn).get(0).outerHTML}</td>`,
```

Comment:
HTML in JS 😐 ?

Reply:
Tried to avoid as much as possible, but once again, idk how to achieve it else way. :disappointed: